src/Server/Query.elm

```diff
@@ -71,16 +72,34 @@ ingredientListParser key ingredients =
         )
 
 
-ingredientParser : List FoodProcess.Process -> String -> Result String Recipe.IngredientQuery
+ingredientParser : List Ingredient.Ingredient -> String -> Result String BuilderQuery.IngredientQuery
 ingredientParser ingredients string =
     case String.split ";" string of
-        [ code, mass ] ->
-            Ok Recipe.IngredientQuery
-                |> RE.andMap (foodProcessCodeParser ingredients code)
+        [ id, mass ] ->
+            let
+                ingredient =
+                    id
+                        |> IngredientID.fromString
+                        |> Ingredient.findByID ingredients
+            in
+            Ok BuilderQuery.IngredientQuery
+                |> RE.andMap (Result.map .id ingredient)
+                |> RE.andMap (Result.map .name ingredient)
+                |> RE.andMap (validateMass mass)
+                |> RE.andMap (Ok BuilderQuery.Default)
+
+        [ id, mass, variant ] ->
+            let
+                ingredient =
+                    id
+                        |> IngredientID.fromString
+                        |> Ingredient.findByID ingredients
```

Comment:
Small note as a nit: it's really idiomatic to have the data source at the last position in the args list, eg here I'd really really expect we have 

```elm
ingredient =
    ingredients
        |> Ingredient.findByID id
```

Maybe there's a good reason doing it the other way?